internet-programming/hw2/client/Client.java

```diff
@@ -0,0 +1,71 @@
+package hw2.client;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.AsynchronousSocketChannel;
+import java.nio.channels.CompletionHandler;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+
+public class Client {
+
+    private static final String SERVER_ADDRESS = "localhost";
+    private static final int PORT = 31337;
+    // private static final int BUFFER_SIZE = 1024;
+
+    public static void main(String[] args) throws IOException, InterruptedException {
+        Path filePath = Paths.get("file.txt");
+        if (!Files.exists(filePath)) {
+            System.out.println("File does not exist: " + filePath);
+            return;
+        }
+
+        AsynchronousSocketChannel clientChannel = AsynchronousSocketChannel.open();
+        Future<Void> future = clientChannel.connect(new java.net.InetSocketAddress(SERVER_ADDRESS, PORT));
+
+        // Wait for the connection to be established
+        
+
+        try {
+            future.get();
+        } catch (ExecutionException e) {
+            throw new RuntimeException(e);
+        }
+
+        sendFile(clientChannel, filePath);
+    }
+
+    private static void sendFile(AsynchronousSocketChannel clientChannel, Path filePath) throws IOException {
+        byte[] fileData = Files.readAllBytes(filePath);

```

Comment:
The entire file is read into memory and if the file is large may rezult to oOM.Better reading the file in chunks using buffer with fixed size.